encoding/int/encoding.go

```diff
@@ -0,0 +1,46 @@
+package int
+
+import "io"
+
+// Encoder encodes int values.
+type Encoder interface {
+	// Encode a collection of ints.
+	// Callers should explicitly call `Reset` before
+	// subsequent call to `Encode`.
+	Encode(Iterator) error
+	// Bytes returns the encoded bytes of the encoder.
+	// The format of the byte slice is as follows:
+	//   - version:                 1 byte
+	//   - metadata length bytes:   VInt
+	//   - metadata bytes:          Variable bytes
+	//   - data bytes:              Variable bytes
+	Bytes() []byte
```

Comment:
Are the comments still applicable here?

Reply:
Yes, they are. The encoded bytes still needs to contain metadata info useful for the decoding process.

Reply:
I'd remove it from the API comments. Feels like an implementation detail to me.

Reply:
👍